gateway/src/test/java/io/camunda/zeebe/gateway/grpc/GrpcErrorMapperTest.java

```diff
@@ -167,4 +169,28 @@ void shouldLogIllegalArgumentExceptionOnDebugWithBigIntTooLong() {
       assertThat(event.getLevel()).isEqualTo(Level.DEBUG);
     }
   }
+
+  @Test
+  void shouldLogInvalidTenantRequestException() {
+    // given
+    final String invalidTenantId = "tenant!@#";
```

Comment:
👍 I like it!

❌ While the used regex is clear, it allows for some "less readable" tenant ids. Personally, I think that's fine, but we should make that choice explicit. Please add some more unit tests to verify the edge cases of valid tenant ids:
- some failing ones like an empty string, just a space, more than 31 chars, etc
- some passing ones like a bunch of dots, a dot and a dash, etc.

Reply:
Nice! I'll create parameterized tests for all of these cases and put them in a dedicated "tenant" test class.